ethereum/core/src/main/java/tech/pegasys/teku/core/StateTransition.java

```diff
@@ -77,52 +76,52 @@ public BeaconState initiate(
         preState,
         signed_block,
         validateStateRootAndSignatures,
-        interimState -> {},
         IndexedAttestationProvider.DIRECT_PROVIDER);
   }
 
   public BeaconState initiate(
       BeaconState preState,
-      SignedBeaconBlock signed_block,
+      SignedBeaconBlock signedBlock,
       boolean validateStateRootAndSignatures,
-      final Consumer<BeaconState> beaconStateConsumer,
       final IndexedAttestationProvider indexedAttestationProvider)
       throws StateTransitionException {
     try {
-      BlockValidator blockValidator =
-          validateStateRootAndSignatures ? this.blockValidator : BlockValidator.NOOP;
-      final BeaconBlock block = signed_block.getMessage();
-
       // * Process slots (including those with no blocks) since block
       // * beaconStateConsumer only consumes the missing slots here,
       //   the new block will be processed when adding to the store.
-      BeaconState postSlotState =
-          process_slots(
-              preState,
-              block.getSlot(),
-              state -> {
-                if (!state.getSlot().equals(block.getSlot())) {
-                  beaconStateConsumer.accept(state);
-                }
-              });
+      BeaconState postSlotState = process_slots(preState, signedBlock.getMessage().getSlot());
```

Comment:
Yeah that was the way I went initially and it does work but makes me very nervous that it would accept a block with the same slot as its parent.  Having an explicit method felt safer as it made it clear this was actually only a part of the state transition in the spec.

That said, I'm really not liking the way `StateTransition` class is panning out - just not sure exactly what to do with it.